src/System.Private.CoreLib/src/System/RtType.cs

```diff
@@ -4857,17 +4848,20 @@ internal unsafe MdUtf8String(void* pUtf8String, int cUtf8String)
             m_StringHeapByteLength = cUtf8String;
         }
 
+        // Very common called version of the Equals pair
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
         internal unsafe bool Equals(MdUtf8String s)
         {
+            bool isEqual = false;
             if (m_pStringHeap == null)
             {
-                return s.m_StringHeapByteLength == 0;
+                isEqual = s.m_StringHeapByteLength == 0 ? true : false;
             }
             if ((s.m_StringHeapByteLength == m_StringHeapByteLength) && (m_StringHeapByteLength != 0))
             {
-                return EqualsCaseSensitive(s.m_pStringHeap, m_pStringHeap, m_StringHeapByteLength);
+                isEqual = SpanHelpers.SequenceEqual<byte>(ref Unsafe.AsRef<byte>(s.m_pStringHeap), ref Unsafe.AsRef<byte>(m_pStringHeap), m_StringHeapByteLength) ? true : false;
```

Comment:
This does not need to use `Unsafe.AsRef`. Regular cast to `byte*` is enough. Maybe change `m_pStringHeap` to `byte*` - it may result into fewer casts.

Reply:
What's the syntax here? Do I need to `ref` the dereference e.g. `ref *m_pStringHeap` it doesn't seem happy using `m_pStringHeap` directly

```
Error	CS1620	Argument 2 must be passed with the 'ref' keyword
```
Adding just ref complains about its type (after complaining a readonly value can't be passed by `ref`)
```
Error	CS1503	Argument 2: cannot convert from 'ref byte*' to 'ref byte'	
```
Using ref deference seems ok, if a bit weird